client.go

```diff
@@ -181,7 +177,7 @@ func (c *Client) PostForm(ctx context.Context, endpoint string, vals url.Values,
 
 	return c.RoundTrip(func() (*http.Response, error) {
 		if len(files) == 0 {
-			req, err := http.NewRequest(http.MethodPost, endpoint, strings.NewReader(vals.Encode()))
+			req, err := http.NewRequest(method, endpoint, strings.NewReader(vals.Encode()))
```

Comment:
nit: if we used `http.NewRequestContext` we could get rid of the need to use `req.WithContext`